### ISSUE TITLE
Potential simpler alternative to ctypes

### ISSUE THREAD

**Issue by rdaly525** [leonardt/pycoreir#48]
https://eli.thegreenplace.net/2013/03/09/python-ffi-with-ctypes-and-cffi/

**Comment by leonardt** [comment]
Yes, we noticed this earlier when considering pypy support, cffi is developed by the pypy team so it has good integration, also I think it provides some mechanisms for dealing with the GIL issues which would simplify our libcoreir-python
